core-common/src/main/java/eu/arrowhead/common/processor/NetworkAddressPreProcessor.java

```diff
@@ -0,0 +1,168 @@
+/********************************************************************************
+ * Copyright (c) 2021 AITIA
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   AITIA - implementation
+ *   Arrowhead Consortia - conceptualization
+ ********************************************************************************/
+
+package eu.arrowhead.common.processor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import eu.arrowhead.common.Utilities;
+
+@Component
+public class NetworkAddressPreProcessor {
+	
+	private final Logger logger = LogManager.getLogger(NetworkAddressPreProcessor.class);
+	
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public String normalize(final String address) {
+		logger.debug("normalize started...");
+		
+		if (Utilities.isEmpty(address)) {
+			return "";
+		}
+		final String candidate = address.trim();
+		
+		// Simple string
+		if (!candidate.contains(".") && !candidate.contains(":")) {
+			return candidate;
+		
+		// Possible IPv4 or domain name
+		} else if(candidate.contains(".") && !candidate.contains(":")) {
+			return candidate;
+		
+		// Possible IPv6
+		} else if (!candidate.contains(".") && candidate.contains(":")) {
+			return normalizeIPv6(candidate);
+			
+		// Possible IPv6-IPv4 hybrid
+		} else {
+			return normalizeIPv6IPv4Hybrid(candidate);
+		}
+	}
+	
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
```

Comment:
minor issue: maybe put some comments which describing the cases, like add leading zeroes, etc,